src/commonComponents/sidebar/Sidebar.js

```diff
@@ -0,0 +1,84 @@
+// react
+import { useEffect, useState } from "react";
+
+// prop-types
+import PropTypes from "prop-types";
+
+//lodash
+import _map from "lodash/map";
+import _startCase from "lodash/startCase";
+
+// constants
+import { INITIAL_INDEX, SCROLL_FACTOR } from "./constants/sidebar.general";
+
+//helpers
+import { sanitizeText } from "../../utility/sanitizeText";
+import { getMatchedIndexWithUrlHash } from "./helpers/sidebar.general";
+
+// components
+import LinkItem from "../linkItem";
+
+// css
+import "./sidebar.css";
+
+const Sidebar = (props) => {
+  const { links, className } = props;
+
+  const [activeIdx, setActiveIdx] = useState(INITIAL_INDEX);
+
+  useEffect(() => {
+    const setInitialActiveIdx = () => {
+      const index = getMatchedIndexWithUrlHash(links);
+      setActiveIdx(index);
+    };
+
+    setInitialActiveIdx();
+  }, [links]);
+
+  const onClick = (index) => {
+    return function handleClick() {
```

Comment:
you are creating a new instance of this function for every link item, you can actually send this handle click function and send the indx from the call site but again its debatable that it would reduce the reusability of link item